scrapy/spiderloader.py

```diff
@@ -18,6 +18,16 @@ class SpiderLoader:
     """
 
     def __init__(self, settings):
+        self.require_name = settings.getbool('SPIDER_LOADER_REQUIRE_NAME')
+        if self.require_name:
+            message = (
+                'SPIDER_LOADER_REQUIRE_NAME is True. In a future version of '
+                'Scrapy, the SPIDER_LOADER_REQUIRE_NAME setting will be '
+                'removed, and Scrapy will always behave as if '
+                'SPIDER_LOADER_REQUIRE_NAME were False. To remove this '
+                'warning, set SPIDER_LOADER_REQUIRE_NAME to False.'
+            )
+            warnings.warn(message, ScrapyDeprecationWarning)
```

Comment:
I don't have a strong opinion on that, but for robots.txt enable option we did a different thing:

* keep old default (robotx.txt handling diisabled)
* set new default in generated setting.py, so that new projects get a new default
* try to document carefully these default values - technically, default is False, but users should expect this option to be True
* the setting itself is not deprecated

Here the approach is different; the goal is to move to a new way of defining spiders, not to provide an alternative.
Do you think there are cases where the current behavior is preferrable? I wonder if we should try moving to the new approach like we did for robots.txt first, and then decide if we should deprecate the current approach or not. Or is the new approach clearly better than the old one?

Reply:
I see no reason why we cannot maintain both behaviors. I’ll make the required changes.